File: app/src/main/java/io/github/kutaiba_nezar_kashmar/multiverse_of_geeks/model/domain/response/CommentResponse.java
package io.github.kutaiba_nezar_kashmar.multiverse_of_geeks.model.domain.response;

import java.util.ArrayList;

import io.github.kutaiba_nezar_kashmar.multiverse_of_geeks.model.domain.Comment;
import io.github.kutaiba_nezar_kashmar.multiverse_of_geeks.model.domain.Commenter;

public class CommentResponse
{
  private String id;
  private String author;
  private Commenter author_details;
  private String content;
  private String created_at;
  private ArrayList<Comment> results = null;

  public Comment getComment()
  {
    return new Comment(id, author, author_details, content, created_at);
  }

  public ArrayList<Comment> getResults()
  {
    return results;
  }
}

File: app/src/main/java/io/github/kutaiba_nezar_kashmar/multiverse_of_geeks/model/domain/response/CommentResponse.kt
package io.github.kutaiba_nezar_kashmar.multiverse_of_geeks.model.domain.response

import io.github.kutaiba_nezar_kashmar.multiverse_of_geeks.model.domain.Comment
import io.github.kutaiba_nezar_kashmar.multiverse_of_geeks.model.domain.Commenter
import java.util.ArrayList

class CommentResponse
{
    val id : String = ""
    val author : String = ""
    val author_details : Commenter = Commenter("")
    val content : String  = ""
    val created_at : String  = ""
    val results : List<Comment> = ArrayList()

    fun getComment() : Comment
    {
        return Comment(id, author, author_details, content, created_at)
    }
}